snapcraft/models/project.py

```diff
@@ -426,6 +445,54 @@ def _validate_default_provider(cls, default_provider):
         return default_provider
 
 
+class Platform(pydantic.BaseModel):
+    """Snapcraft project platform definition."""
+
+    build_on: list[SnapArch] | None = pydantic.Field(min_items=1, unique_items=True)
+    build_for: list[SnapArch] | None = pydantic.Field(min_items=1, unique_items=True)
+
+    class Config:  # pylint: disable=too-few-public-methods
+        """Pydantic model configuration."""
+
+        allow_population_by_field_name = True
+        alias_generator: Callable[[str], str] = lambda s: s.replace(  # noqa: E731
+            "_", "-"
+        )
+
+    @pydantic.validator("build_for", pre=True)
+    @classmethod
+    def _vectorise_build_for(cls, val: str | list[str]) -> list[str]:
+        """Vectorise target architecture if needed."""
```

Comment:
I like this. It allows us to accept just a string in `build-for`, but when we generate a json schema it'll complain if it's not a list